app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -173,14 +168,35 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         return START_STICKY;
     }
 
+    private static boolean shouldServiceRun() {
+        if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2) return false;
+        final boolean result = (DexCollectionType.hasXbridgeWixel() || DexCollectionType.hasBtWixel()) && !Home.get_forced_wear();
+        Log.d(TAG, "shouldServiceRun() returning: " + result);
+        return result;
+    }
+
     @Override
     public void onDestroy() {
         status("Shutdown");
         super.onDestroy();
         Log.d(TAG, "onDestroy entered");
         close();
         foregroundServiceStarter.stop();
-        setRetryTimer();
+        if (shouldServiceRun()) {//Android killed service
+            setRetryTimer();
+        }
+        else {//onDestroy triggered by CollectionServiceStart.stopBtService
+            if (serviceIntent != null) {
```

Comment:
I think we should wrap this with a try/catch block because what if `alarm.cancel()` throws an exception for example. I don't know under what circumstances it could but perhaps with a race condition between the alarm firing and us trying to cancel it.